### PR TITLE
Add environment variable setters to cloudsdk builder

### LINKED ISSUE #39: make sure we allow setting the environment variable to identifying the deployment source

### PR DESCRIPTION
**Pull request by etanshaul**
fixes #39 

@loosebazooka @meltsufin, thoughts?

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/process/ProcessRunner.java [hunk 1]
```diff
@@ -14,11 +14,15 @@
...
+import java.util.Map;
+
 /**
  * Interface for running arbitrary commands.
  */
 public interface ProcessRunner {
 
   void run(String[] command) throws ProcessRunnerException;
 
+  void run(String[] command, Map<String, String> environment) throws ProcessRunnerException;
```
Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/process/ProcessRunner.java]: I'm not sure about overloading the run method. Maybe we can just add a `setEnvironment(Map<String, String> environment)`?

Comment (etanshaul) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/process/ProcessRunner.java]: Yea, I think thats ok too. I'll update it

**Comment by etanshaul** [comment]
@meltsufin removed overloading and added a setter

**Comment by meltsufin** [comment]
Looks good.
...All these members with appCommand* prefix are starting to make CloudSdk look like its trying to do too much, but that's a thing we should deal with separately.

**Comment by etanshaul** [comment]
I agree. I actually have one more to add based on @chanseokoh on the issue. It might be worthwhile (in a different PR) to extract those out into a separate builder object that's passed in to the sdk.

**Comment by etanshaul** [comment]
Added support for the metrics environment version env var.

**Comment by meltsufin** [comment]
LGTM

**Pull request merged**